inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
-                                    request_id text,
-                                    cron_duration text,
-                                    cron_minutes text,
-                                    cron_hours text,
-                                    cron_day_month text,
-                                    cron_month text,
-                                    cron_day_week text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
+    def _create_repeated_schedule(self, rs: RepeatedSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        sql = ''' INSERT INTO repeated_schedule(request_id, cron_duration, cron_minutes, cron_hours, cron_day_month, cron_month, cron_day_week)
+                                VALUES(?,?,?,?,?,?,?); '''
+        try:
+            self._cursor.execute(sql, (rs.request_id, 
+                                        rs.cron_duration, 
+                                        rs.cron_minutes, 
+                                        rs.cron_hours, 
+                                        rs.cron_day_month, 
+                                        rs.cron_month, 
+                                        rs.cron_day_week))
+        except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+            raise DispatcherException(f"Error adding repeated schedule to database: {e}")
+        
+        schedule_id = self._cursor.lastrowid
+        if not schedule_id:
+            raise DispatcherException("No schedule id was added to the repeated_schedule table.")
+        logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}") # noqa
+        
+        # Add the manifests to the manifest table
+        manifest_ids = self._insert_manifest_to_table(rs.manifests)
+           
+        # Add the schedule_id and manifest_id to the repeated_schedule_manifest table
+        self._insert_repeated_schedule_manifest_tables(schedule_id, manifest_ids)
+        self._conn.commit()
+        
+    def _insert_manifest_to_table(self, manifests: list[str]) -> list[int]:
+        # Add the manifest to the manifest table
+        manifest_ids: list[int] = []
+        
+        if len(manifests) == 0:
+            raise DispatcherException("Error: At least one manifest is required for the schedule.  Manifests list is empty.")
+        
+        for manifest in manifests:
+            sql = ''' INSERT INTO manifest(manifest) VALUES(?); '''
+            try: 
+                self._cursor.execute(sql, (manifest,))
+            except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Optimize the manifest insertion by using executemany instead of executing insert statements in a loop.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        sql = ''' INSERT INTO manifest(manifest) VALUES(?); '''
        manifest_tuples = [(manifest,) for manifest in manifests]
        self._cursor.executemany(sql, manifest_tuples)
```

Reply:
This would be a good idea, but in this case I need the id of each manifest as it is added for the join table.